### PR TITLE
Spessmart

### PR DESCRIPTION
**Pull request by Shadowmech88**
Pulls #8949 out of the grave.
All references to "Costco" changed to "Spessmart" for immulsions' sake.
Original PR text:

image

Added a huge store that can appear somewhere in space. It's operated by robots!
In the store you can buy expensive things for cheap, and very rare things (switchtools, battering rams, tank transfer valves, ...) for lots of money. You can also buy circuitboards (75$ each) or clothing (150$ each). Stock is randomly generated every round

How to buy something:
buying_at_spessmart

It's guarded by lots of extremely robust and fast robots with 200 health, built-in rifle and 15-25 melee damage. Crossing the warning stripes while holding an unpaid item will [instantly lock down the entire store and send the guard bots after you.](http://puu.sh/nNMxh/9a25a8f05a.jpg)

If you don't have any money, you can get a drink at the bistro - alcohol and water is free there.

OTHER CHANGES:

* IDs on corpse landmarks now properly shows occupation
* existing_typesof now performs an additional check

:cl:
 * rscadd: Added a new vault, a large supermarket operated by robots, courtesy of unid15.

### CODE REVIEW

**Comment by Lokiamis** [comment]
Requesting salvage ship dock if there isn't one

**Comment by Shadowmech88** [comment]
I'm not sure how that would work, allowing the salvage shuttle to dock at a location that isn't always present and never in the same place.

Also the salvage shuttle is always in the same place, which would make it too easy to find the vault, I think. Just find the salvage shuttle first and it would take you right to the vault if it exists.

**Comment by Lokiamis** [comment]
Oh, right. Didn't realize it wasn't static.

**Comment by Aurx** [comment]
List of all the items for sale please, for the sake of discussion.
Also, why is this PR touching lance.dm?  Atomic, man, atomic.

**Comment by Shadowmech88** [comment]
I just copied all the code from unid15's PR. No idea why *he* touched `lance.dm`.

**Comment by kilozombie** [comment]
Having the items be random _and_ random whether or not the vault appears might make it a little rare to see use. But this is fun. I can imagine spess voyages being even more interesting if you have a few hundred creds on you.

**Comment by unid15** [comment]
Thank you!

* You can discard the changes in lance.dm, they aren't necessary

* List of items for sale is here https://github.com/Shadowmech88/vgstation13/blob/8e979ecb3781d43e1fbeadabf0641470621f730c/maps/randomvaults/spessmart.dm

**Comment by Shadowmech88** [comment]
* Changes to lance.dm discarded.

**Comment by ComicIronic** [comment]
Still got layer edits in the map, which we don't want.

**Comment by Shadowmech88** [comment]
I'm not sure what that means, so tell me what to Ctrl+F for and remove.

**Comment by ComicIronic** [comment]
Any case where you still see `layer = ` is a manual layer setting. Layer is the var which controls the order stuff is drawn, so if two atoms are on the same tile, the one with higher layer shows above.

**Comment by Shadowmech88** [comment]
Is there anything it can be replaced with, like `pixel_x` for `step_x`, or does it just need to be removed?

**Comment by ComicIronic** [comment]
You just need to remove it.

**Comment by ihadtoregisterforthis** [comment]
Use the fancy shuttle destination disks to allow returning to the Spessmart with a ship

**Comment by Shadowmech88** [comment]
* Spessmart now has a docking port at one of its entrances.
* The first Spessmart customer will receive a Spessmart shuttle destination disk from the cashier free with their first purchase.

File: maps/randomvaults/spessmart.dm [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+/mob/living/simple_animal/robot/robot_cashier/attack_hand(mob/user)
+	if(user.a_intent == I_HELP)
+		if(world.time < help_cd + 0.5 SECONDS)
+			return
+
+		spawn(3)
+			help_cd = world.time
+			var/area/vault/supermarket/shop/shop = get_area(src)
+			if(!istype(shop))
+				say("ERROR> :$$-UAable to DoAAect to the dostDo maiAframe-%OI51")
```
Comment (unid15) [inline maps/randomvaults/spessmart.dm:341]: kind of a reference to costco

Comment (Probe1) [reply maps/randomvaults/spessmart.dm:341]: It's not too overt and it makes it kind of interesting

File: code/game/objects/effects/portals.dm [hunk 2]
```diff
@@ -47,6 +47,15 @@
 /obj/effect/portal/New(turf/loc,var/lifespan=300)
 	..()
 	playsound(loc,'sound/effects/portal_open.ogg',60,1)
+
+	var/area/A = get_area(src)
+	if(A.flags & NO_PORTALS)
+		spawn()
```
Comment (N3X15) [inline code/game/objects/effects/portals.dm]: This spawn looks a little sketchy and unnecessary.

Comment (Shadowmech88) [reply code/game/objects/effects/portals.dm]: I could remove it if you think it'd be best.

**Pull request merged**